client/state/themes/selectors/get-is-live-preview-supported.ts

```diff
@@ -151,11 +148,23 @@ export const getIsLivePreviewSupported = ( state: AppState, themeId: string, sit
 
 		/**
 		 * Disable Live Preview for themes that are NOT included in a plan.
-		 * This should be updated as we implement the flow for them.
-		 * Note that BTP works on Atomic sites if a theme is installed.
-		 * @see https://github.com/Automattic/wp-calypso/issues/79223
 		 */
 		if ( ! canUseTheme( state, siteId, themeId ) ) {
+			/**
+			 * Enable Live Preview for Premium and WooCommerce themes.
+			 * @see https://github.com/Automattic/wp-calypso/issues/79223
+			 */
+			if (
+				config.isEnabled( 'themes/block-theme-previews-premium-and-woo' ) &&
+				( isThemePremium( state, themeId ) || isThemeWooCommerce( state, themeId ) )
```

Comment:
Thanks for reaching out, @renatho!

This feature should be compatible with all Bundle themes in the future (CC: @Automattic/lego). However, several changes are needed to transition it from being Woo themes-specific to being applicable to Bundle themes. For example;

- The logic to decide if the site has the feature to use the theme: 
	- https://github.com/Automattic/wp-calypso/blob/c2288803b8ab3a3f04444de6f12bdfdb6ede9628/apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-can-preview-but-need-upgrade.ts#L30-L34
	- Is this condition the same for all Bundle themes?
- The plan they need: 
	- https://github.com/Automattic/wp-calypso/blob/c2288803b8ab3a3f04444de6f12bdfdb6ede9628/apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-can-preview-but-need-upgrade.ts#L119-L126
	- Is this condition the same for all Bundle themes?
- The display name of a theme type: 
	- https://github.com/Automattic/wp-calypso/blob/c2288803b8ab3a3f04444de6f12bdfdb6ede9628/apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-previewing-theme.ts#L47
	- Can this be retrieved by `useBundleSettings`?

I would greatly appreciate it if you could implement all the changes simultaneously. But, if this isn't feasible, please leave it as it is, and I will file an issue to address it later. 🙂 

For functionality testing: pekYwv-3xz-p2

Reply:
Hi @okmttdhr! 👋

> please leave it as it is, and I will file an issue to address it later

After checking this a little, I think it would be the best approach. And if you want to reference me in the issue, I can help with what is needed regarding the bundle feature.

I think the conclusion (not concluded yet) of the thread https://github.com/Automattic/wp-calypso/pull/84886#discussion_r1427081784 will be a good reference for the needs here.

Reply:
📝 https://github.com/Automattic/wp-calypso/issues/86274